src/app/modules/employee/employee.service.ts

```diff
@@ -0,0 +1,40 @@
+import { Injectable } from '@nestjs/common';
+
+import { BcryptService } from '../../../common/bcrypt.service';
+import { ValidationService } from '../../../common/validation.service';
+import Conflict from '../../../core/exceptions/conflict';
+import { EmployeeRepository } from '../../../database/repositories/employee.repository';
+import * as dto from './employee.dto';
+import { CreateEmployeeRequestSchema } from './employee.validation';
+
+@Injectable()
+export class EmployeeService {
+  constructor(
+    private employeeRepository: EmployeeRepository,
+    private validationService: ValidationService,
+    private bcyptService: BcryptService,
+  ) { }
+
+  async createEmployee(
+    data: dto.CreateEmployeeRequest
+  ): Promise<dto.CreateEmployeeResponse> {
+    const bodyReq = await this.validationService.validate(CreateEmployeeRequestSchema, data);
+
+    const dbEmployee = await this.employeeRepository.findByUsername(bodyReq.username);
+
+    if (dbEmployee) {
+      throw new Conflict('Username already used. Please use another username.', 'Conflict error');
+    }
+
+    const hashedPassword = await this.bcyptService.hash(bodyReq.password);
+    bodyReq.password = hashedPassword;
```

Comment:
Bisa dilangsung ini keknya

```suggestion
    bodyReq.password = await this.bcyptService.hash(bodyReq.password);    
```